src/schema.ts

```diff
@@ -10,11 +10,22 @@ import { capitalize, writeFile } from "./utils";
 const compileOptions: Partial<CompilerOptions> = { bannerComment: "" };
 const defaultSchema = { type: "object", additionalProperties: false };
 
-function addDefaultValueToSchema(schema: any) {
-  return {
-    ...schema,
-    additionalProperties: schema.additionalProperties || false,
-  };
+export function addDefaultValueToSchema(
+  schema: Record<string, any>
+): Record<string, any> {
+  return Object.entries(schema).reduce((acc, [key, value]) => {
+    if (value !== null && typeof value == "object") {
```

Comment:
I think there needs to be a way of ignoring Arrays, as otherwise those will be mapped to an object. This isn't pretty, but something like
``` 
if (
      value !== null &&
      typeof value == "object" &&
      Array.isArray(value) === false
    )
```